### DISCUSSION TITLE
PDF to CSV?

### DISCUSSION THREAD

**Discussion by ZacharyJSmit** [firefly-iii/firefly-iii#6082] [General]
My banks don't connect to the nice import methods.  I need to download a PDF, convert to CSV and then upload to firefly.  There are a lot of online converters, but most take a line of text and place it into a single cell, instead of creating columns.  What do you guys use to convert bank PDFs to CSV?

**Comment by JC5** [comment]
PDF to anything is a nightmare. Can you even select text from the PDF file or is it an image?

**Reply by Zyadon7** [reply]
I can select the text, but copying into excel does not create columns in a usable way.  Most of the pdf to csv tools I've looked at so far give me similar problems.  I'm fine with cleaning up enough to upload to the importer, but I'm hoping to find an initial converter that at least separates a table into columns correctly.

**Reply by top-cg** [reply]
This is not open source but i once tried it to convert amex pdf files into csv : https://www.propersoft.net/products/transactions

**Reply by Zyadon7** [reply]
Tabula has been the best free converter I've found so far.  If you highlight areas manually, it's able to discern columns in tables.  It's a pretty manual process and I'd love something smarter, but it works and it is much better than entering everything manually.

**Reply by hiyan** [reply]
Pasting it into chatgpt and asking it to 'clean it up' works for me.

**Comment by Rocketplanner83** [comment]
Has this been improved on in 2 years??  It would be nice if fireflii could do this itself

**Reply by JC5** [reply]
No, and this will not be a feature I'm afraid. Sorry about that.

**Reply by Rocketplanner83** [reply]
Okay, how about syncing to banks and brokerage houses in USA?
